src/mobilenet.jl

```diff
@@ -0,0 +1,296 @@
+"""
+function mobilenetv3(configs, width_mult; max_width = 1024, nclasses = 1000)
+  # building first layer
+  inplanes = _make_divisible(16 * width_mult, 8)
+  layers = []
+  append!(layers, conv_bn((3, 3), 3, inplanes, hardswish; stride = 2))
+  explanes = 0
+  # building inverted residual blocks
+  for (k, t, c, use_se, use_hs, s) in configs
+    # inverted residual layers
+    outplanes = _make_divisible(c * width_mult, 8)
+    explanes = _make_divisible(inplanes * t, 8)
+    push!(layers, invertedresidualv3(inplanes, explanes, outplanes, k, s, use_se, use_hs))
+    inplanes = outplanes
+  end
+
+  # building last several layers
+  output_channel = max_width
+  output_channel = width_mult > 1.0 ? _make_divisible(output_channel * width_mult, 8) : output_channel
+  classifier = (
+    Dense(explanes, output_channel, hardswish),
+    Dropout(0.2),
+    Dense(output_channel, nclasses),
+  )
+
+  return Chain(Chain(layers...,
+                     conv_bn((1, 1), inplanes, explanes, hardswish, bias = false)...),
+               Chain(AdaptiveMeanPool((1, 1)), flatten, classifier...))
+end
+
+# Configurations for small and large mode for MobileNetv3
+mobilenetv3_configs = Dict(
+  :small => [
+    # k, t, c, SE, HS, s 
+    (3, 1, 16, true, false, 2),
+    (3, 4.5, 24, false, false, 2),
+    (3, 3.67, 24, false, false, 1),
+    (5, 4, 40, true, true, 2),
+    (5, 6, 40, true, true, 1),
+    (5, 6, 40, true, true, 1),
+    (5, 3, 48, true, true, 1),
+    (5, 3, 48, true, true, 1),
+    (5, 6, 96, true, true, 2),
+    (5, 6, 96, true, true, 1),
+    (5, 6, 96, true, true, 1),
+  ], 
+  :large => [
+    # k, t, c, SE, HS, s 
+    (3, 1, 16, false, false, 1),
+    (3, 4, 24, false, false, 2),
+    (3, 3, 24, false, false, 1),
+    (5, 3, 40, true, false, 2),
+    (5, 3, 40, true, false, 1),
+    (5, 3, 40, true, false, 1),
+    (3, 6, 80, false, true, 2),
+    (3, 2.5, 80, false, true, 1),
+    (3, 2.3, 80, false, true, 1),
+    (3, 2.3, 80, false, true, 1),
+    (3, 6, 112, true, true, 1),
+    (3, 6, 112, true, true, 1),
+    (5, 6, 160, true, true, 2),
+    (5, 6, 160, true, true, 1),
+    (5, 6, 160, true, true, 1)
+  ]
+)
+
+# Model definition for MobileNetv3
+struct MobileNetv3
+  layers
+end
+
+"""
+    MobileNetv3(mode::Symbol = :small, width_mult = 1.0; pretrain = false, nclasses = 1000)
+
+Create a MobileNetv3 model with the specified configuration.
+([reference](https://arxiv.org/abs/1905.02244)).
+Set `pretrain = true` to load the model with pre-trained weights for ImageNet.
+
+# Arguments
+- `mode`: :small or :large for the size of the model (see paper).
+- `width_mult`: Controls the number of output feature maps in each block
+                (with 1.0 being the default in the paper).
+- `pretrain`: whether to load the pre-trained weights for ImageNet
+- `nclasses`: the number of output classes
+
+See also [`Metalhead.mobilenetv3`](#).
+"""
+function MobileNetv3(mode::Symbol = :small, width_mult = 1.0; pretrain = false, nclasses = 1000)
```

Comment:
```suggestion
function MobileNetv3(mode = :small, width_mult = 1; pretrain = false, nclasses = 1000)
```

Reply:
This one runs into problems when I remove the ::Symbol. It doesn't call the function and just displays as `MobileNetv3(:small)`. Hence the type annotation. Same comment for the `width_mult` as above

Reply:
Oh right cause it conflicts with the default constructor for the struct. We can leave them in then, but widen the type of `widen_mult` to `Number` instead of `Float64`.